src/persist-client/src/internal/state_diff.rs

```diff
@@ -682,7 +684,7 @@ fn apply_diffs_spine<T: Timestamp + Lattice>(
     for (batch, ()) in batches {
         // Ignore merge_reqs because whichever process generated this diff is
         // assigned the work.
-        let _merge_reqs = new_trace.push_batch(batch);
+        let _merge_reqs = new_trace.push_batch_no_merge_reqs(batch);
```

Comment:
I did leave the comments in on purpose, thinking of them as a justification for why we're calling `push_batch_no_merge_reqs` instead of `push_batch`. ymmv of course